metals/src/main/scala/scala/meta/internal/metals/ScalacDiagnostic.scala

```diff
@@ -1,9 +1,18 @@
 package scala.meta.internal.metals
 
+import scala.jdk.CollectionConverters._
+
 import org.eclipse.{lsp4j => l}
 
 object ScalacDiagnostic {
 
+  object ScalaAction {
+    def unapply(d: l.Diagnostic): Option[l.DiagnosticRelatedInformation] =
+      if (d.getSource == "scala-cli")
```

Comment:
I wonder if we might want to enable it for more build tools?

CC @ckipp01 This will work only for ScalaCLI for now, but I think this fits into the better diagnostics idea, what do you think about it? We could expand it to sbt/bloop/mill since they don't currently fill in that field.

```suggestion
      if (d.getRelatedInformation() != null && d.getRelatedInformation().nonEmpty)
```